Add: a simple smoke test over localhost

larswirzenius opened the pull request.

Commit:
Add: a simple smoke test over localhost

rjek commented on Makefile:
```diff
@@ -6,3 +6,6 @@ fling: fling.c
 clean:
 	rm -f fling
+
+check:
```
The check target should probably depend on the fling target?

larswirzenius replied:
Good point. I'll fix.

Commit:
Add: have check target depend on fling

This ensures the fling executable has been built before the test that
uses it runs.
